server/TypeDBServer.java

```diff
@@ -210,66 +226,67 @@ protected void serve() {
             server.awaitTermination();
         } catch (InterruptedException e) {
             // server is terminated
-            close();
-            Thread.currentThread().interrupt();
+            try {
+                close();
+                System.exit(0);
+            } catch (Throwable error) {
+                logger().error("Unexpected error during shutdown, performing hard stop.", error);
+                Runtime.getRuntime().halt(1);
+            }
         }
     }
 
     @Override
-    public synchronized void close() {
-        logger().info("");
-        logger().info("Shutting down {}...", name());
-        try {
-            assert typeDBService != null;
-            typeDBService.close();
-            server.shutdown();
-            server.awaitTermination();
-            databaseMgr.close();
-            System.runFinalization();
-            logger().info("{} has been shutdown", name());
-        } catch (InterruptedException e) {
-            logger().error(FAILED_AT_STOPPING.message(), e);
-            Thread.currentThread().interrupt();
+    public synchronized void close() throws TypeDBCheckedException {
+        if (isOpen.compareAndSet(true, false)) {
+            try {
+                logger().info("");
+                logger().info("Closing {} instance...", name());
```

Comment:
put these into the `try` because they can trigger OOM when right the limit of memory